File: vivid/src/main/java/com/icetea09/vivid/imagepicker/ImagePickerWithActivity.java
package com.icetea09.vivid.imagepicker;

import android.app.Activity;

/**
 * Created by trinhlbk on 2/19/17.
 */

public class ImagePickerWithActivity extends ImagePicker {

    private Activity activity;

    ImagePickerWithActivity(Activity activity) {
        this.activity = activity;
        init(activity);
    }

    @Override
    public void start(int requestCode) {
        activity.startActivityForResult(ImagePickerActivity.newIntent(activity, config), requestCode);
    }
}

File: library/src/main/java/com/icetea09/vivid/imagepicker/ImagePickerWithActivity.kt
package com.icetea09.vivid.imagepicker

import android.app.Activity

/**
 * Created by trinhlbk on 2/19/17.
 */

class ImagePickerWithActivity internal constructor(private val activity: Activity) : ImagePicker() {

    init {
        init(activity)
    }

    override fun start(requestCode: Int) {
        activity.startActivityForResult(ImagePickerActivity.newIntent(activity, config), requestCode)
    }
}
